src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2361,17 +2422,28 @@ object Desugar {
       }
     }
 
+    def removeDuplicates(proofs: Vector[(Type, InterfaceT, st.MethodImpl, st.MethodSpec)]) : Vector[(Type, InterfaceT, st.MethodImpl, st.MethodSpec)] = {
+      val (_, ret) = proofs.foldLeft((Set[(Type, st.MethodImpl, st.MethodSpec)](), Vector[(Type, InterfaceT, st.MethodImpl, st.MethodSpec)]()))(
+        (s: (Set[(Type, st.MethodImpl, st.MethodSpec)], Vector[(Type, InterfaceT, st.MethodImpl, st.MethodSpec)]), proof : (Type, InterfaceT, st.MethodImpl, st.MethodSpec)) =>
+        if (s._1((proof._1, proof._3, proof._4))) s
+        else (s._1 + ((proof._1, proof._3, proof._4)), s._2 ++ Vector(proof))
+      )
+      ret
+    }
+
     lazy val interfaceImplementations: Map[in.InterfaceT, SortedSet[in.Type]] = {
-      info.interfaceImplementations.map{ case (itfT, implTs) =>
+      val firstMap = info.interfaceImplementations.map{ case (itfT, implTs) =>
         (
           interfaceType(typeD(itfT, Addressability.Exclusive)(Source.Parser.Unsourced)).get,
           SortedSet(implTs.map(implT => typeD(implT, Addressability.Exclusive)(Source.Parser.Unsourced)).toSeq: _*)
         )
       }
+      val secondMap = (embeddedInterfaces.groupMap(_._1)(_._2).map(x => (x._1, SortedSet(x._2: _*))): Map[in.InterfaceT, SortedSet[in.Type]])
+      (firstMap.keySet ++ secondMap.keySet).map(k => (k, firstMap.getOrElse(k, SortedSet[in.Type]()) ++ secondMap.getOrElse(k, SortedSet[in.Type]()))).toMap
     }
     def missingImplProofs: Vector[in.Member] = {
 
-      info.missingImplProofs.map{ case (implT, itfT, implSymb, itfSymb) =>
+      removeDuplicates(info.missingImplProofs).map{ case (implT, itfT, implSymb, itfSymb) =>
```

Comment:
This looks hacky. `missingImplProofs` should make sure that it is not returning duplicates and not the caller.